src/gateway/routers/Aggregator.sol

```diff
@@ -34,7 +34,7 @@ contract Aggregator is Auth, IAggregator {
     address[] public routers;
     mapping(address router => Router) public activeRouters;
     mapping(bytes32 messageHash => Message) public messages;
-    mapping(bytes32 messageHash => Recovery) public recoveries;
+    mapping(address router => mapping(bytes32 messageHash => uint256 timestamp)) public recoveries;
```

Comment:
Since the router is part of the mapping now, the value doesn't need to be a struct anymore